api/token_refresh_api.py

```diff
@@ -41,8 +38,9 @@ def validate_token(self, token, email):
 
   # Note: we use only POST instead of GET to avoid attacks that use GETs.
   def do_post(self):
```

Comment:
Is this being called when a session cookie expires, every 10 days?

Reply:
Now that you mention it, I think I probably need to make sure this happens more often.

I added this session refresh to the existing XSRF token refresh requests.  Those happen whenever the user submits a form or causes the UI to do a POST request to the API, unless the XSRF token is less than two hours old.  So, if a user is actually working with the site they are likely to call this API method, but that is not always the case.  For example, they could simply navigate around for 10 days and then they would be suddenly signed out.

I'll add a call to setInterval() to refresh the session daily even if the user never makes any changes.  And, also refresh the session on every navigation.

Reply:
I was only able to do half of what I wanted to do.

I was able to make it refresh the session on each page navigation.  That should be pretty good.  As long as the user does something to cause a new page load with the 10-day window, their session should be refreshed and they should stay signed in.

However, I was not able to make the setInterval() part work.  We use the fetch() method and the session cookie is httpOnly.  Apparently that combination makes it impossible for an API response from the server to set a cookie.  So, if a user opens a form, and sits there 10 days, they won't be able to submit the form.  That's not ideal, but is probably not going to be problem very often, especially since loading a chromestatus page in another tab would set the cookie.

Reply:
I think you might need [`credentials`](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials) header for fetch(), `credentials: "same-origin"`.  It is not set be default https://developer.mozilla.org/en-US/docs/Web/API/fetch#parameters. Regardless, the likelihood that someone leaves a form open for 10 days is small. So LGTM

Reply:
We already set `credentials: "same-origin"`.  I am thinking that the httpOnly aspect of the session cookie is the reason, but I don't want to change that.   We'll see if this is good enough.